packages/sails-ng-common/src/config/component/tab.model.ts

```diff
@@ -0,0 +1,33 @@
+import {FormComponentDefinition} from "../form-component.model";
+import {BaseFormFieldComponentConfig, BaseFormFieldComponentDefinition} from "../form-field-component.model";
+
+
+export interface TabFormFieldComponentDefinition extends BaseFormFieldComponentDefinition {
+    class: "TabComponent";
+    config?: TabComponentConfig;
+}
+
+export interface TabComponentEntryDefinition {
+    id: string; // internal identifier for the tab
+    buttonLabel: string; // The text on the button
+    componentDefinitions: FormComponentDefinition[]; // The components to render in the tab
+    selected?: boolean; // Whether the tab is selected on initialization
+}
+
+export class TabComponentConfig extends BaseFormFieldComponentConfig {
+    mainCssClass?: string; // Main CSS class for the tab component
+    buttonSectionCssClass?: string; // CSS class for the tab buttons
+    tabContentSectionCssClass?: string; // CSS class for the tab content
+    tabPaneCssClass?: string; // CSS class for the tab pane
+    tabPaneActiveCssClass?: string; // CSS class for the active tab pane
+    tabs?: TabComponentEntryDefinition[];
```

Comment:
I guess you're used `tabs` as the config property, and not `componentDefinitions`, because the tab component and tab content component don't have model or layout, only component config?

Is there any chance we'll want to add model or layout config to the tabs in the future?